SpeedGrader - Extract Tabs view from Page view

rh12 opened the pull request:
refs: MBL-18907
affects: Teacher
release note: none

This is just a chore in preparation for [MBL-18907](https://instructure.atlassian.net/browse/MBL-18907).

- Extracted `SpeedGraderPageTab` model
- Extracted `SpeedGraderPageTabsView`
  - it uses the page's viewmodel at the moment, mostly to allow easier rearranging of tab content if needed
- Cleaned up `SpeedGraderPageView`
- Added some helpers to `DrawerState` to make related logic a bit more readable
- Added `onHeightChange()` methods for cases when we only interested in the `height`, not the `size`

## Test plan
- Verify SpeedGrader tab changing, focus, a11y focus, etc. work as before, both in Landscape & Portrait.
- Verify swapping Landscape/Portrait mode works as before, and it keeps the selected tab.

NOTE: Tapping on the selected tab when the drawer is closed still won't open the drawer.

## Checklist
- [x] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [ ] Tested in dark mode
- [x] Tested in light mode

[MBL-18907]: https://instructure.atlassian.net/browse/MBL-18907?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

inst-danger commented:
<details><summary>Teacher Build QR Code:</summary>
<img src="https://api.qrserver.com/v1/create-qr-code/?data=https%3A%2F%2Fapp.bitrise.io%2Fapp%2F88a3641e29e4745c%2Finstallable-artifacts%2F1dd5761e14126a73%2Fpublic-install-page%2F5500b1802e287e2315c168fdc684cbc4&size=150x150" />
</details>


<!-- TeacherBuild -->

inst-danger commented:
<!--
  0 failure: 
  2 warning:  This pull request..., One or more files...
  
  3 markdown notices
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>One or more files are below the minimum test coverage 50%</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher

[MBL-18907](https://instructure.atlassian.net/browse/MBL-18907)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.39% | 91.39% | 0.01%
Core/Core/Common/CommonUI/CoreWebView/View/CoreWebViewAccessibilityHelper.swift | 0% | 0% | 0%
Student/Student/WidgetRouting/WidgetRouter+GradeListWidget.swift | 0% | 0% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c4742cce45f069df147464110325b56faba1b387
</p>

ndrsszsz reviewed (approved):
LGTM

rh12 commented:
@vargaat 

> As an improvement we could resign first responder when changing tabs in landscape mode to have more space for the grades tab without the user manually closing the keyboard (that is alone not trivial in iPhone landscape mode) and also to work around the text suggestion highlight bug of the text editor.

Thanks! I've fixed the issue.
Also fixed VoiceOver autofocusing on split view open.

vargaat reviewed (approved):
QA+1
